src/main/java/com/rackspacecloud/metrics/ingestionservice/listeners/UnifiedMetricsListener.java

```diff
@@ -90,4 +68,16 @@ public void onIdleContainer(Map<TopicPartition, Long> map, ConsumerSeekCallback
     public long getBatchProcessedCount(){
         return batchProcessedCount;
     }
+
+    public static String replaceSpecialCharacters(String inputString){
+        final String[] metaCharacters =
+                {"\\",":","^","$","{","}","[","]","(",")",".","*","+","?","|","<",">","-","&","%"," "};
```

Comment:
Consider pre-compiling a regex pattern into a static variable, then reusing it when making a matcher to use with replaceAll, something like `pattern.matcher(text).replaceAll("_")` ; Pattern is also immutable, so this should remain thread-safe

Reply:
There's also StringUtils.containsAny